tests/unit/test_dns.py

```diff
@@ -13,18 +13,24 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from gordon.interfaces.dns_client import IDNSProviderClient
+
+import asyncio
 
 import pytest  # NOQA
+from gordon import interfaces
 
-from gordon_gcp import CloudDNSClient
+from gordon_gcp import dns
 
 
 def test_implements_interface():
-    """CloudDNSClient provides IDNSProviderClient"""
+    """GDNSPublisher implements IPublisherClient"""
     config = {'foo': 'bar'}
-    client = CloudDNSClient(config)
-    assert issubclass(CloudDNSClient, IDNSProviderClient)
-    assert isinstance(client, IDNSProviderClient)
-    assert config == client.config
-    assert 'Google Cloud DNS' == client.name
+    success, error = asyncio.Queue(), asyncio.Queue()
+    client = dns.GDNSPublisher(config, success, error)
+
+    assert interfaces.IPublisherClient.providedBy(client)
+    assert interfaces.IPublisherClient.implementedBy(dns.GDNSPublisher)
```

Comment:
I don't think I understand how these assertions work - to me it looks like they shouldn't be passing? 

The IEventConsumer interface defined in `gordon.interfaces` specifies that the implementer should have the following methods: update_phase and publish_changes. Its initializer should accept these args: `config, success_channel, error_channel, loop, metrics=None`. However, the GDNSPublisher class sports only the "update_phase" method and accepts the following initializer args: `config, success_channel, error_channel` (no `loop` arg).

Since the GDNSPublisher doesn't have the methods required by the interface and the `__init__` signature is different, I wouldn't expect this to pass. Can you help me understand why it is passing?

Reply:
It only returns `True` because of the decorator on the class, `@zope.interface.implementer(...)` (!!!). Should probably look into/start using zope's [verifying](https://zopeinterface.readthedocs.io/en/latest/verify.html). Mind if I add that later?

Reply:
Ah, makes sense now :). Thanks a lot for looking into this. I'm fine with this getting added in some time later.